generatebundlefile/hack/update_bundles.sh

```diff
@@ -0,0 +1,150 @@
+    .imageDetails
+    | map(select(.imageTags | length > 0))
+    | map(select(.imageTags | map(test($kv)) | any))
+    | sort_by(.imagePushedAt)
+    | last
+    | .imageTags')
+  else
+    output=$(echo $json_output | jq -r '
+    .imageDetails
+    | sort_by(.imagePushedAt)
+    | last(.[]).imageTags')
+  fi
+
+  # Convert tags to array
+  local versions=()
+  for value in $(echo "$output" | jq -r '.[]'); do
+    versions+=("$value")
+  done
+
+  # Get latest version and remove tags with helm
+  local latest_version=$(printf "%s\n" "${versions[@]}" |
+    grep -v "helm" |
+    sort -rV |
+    head -n 1)
+
+  echo $latest_version
+}
+
+# Function to update input bundle file for a given Kubernetes version
+function update_bundle() {
+  k8s_version=$1
+  environment=$2
+  k8s_yaml_file="${BASE_DIRECTORY}/generatebundlefile/data/bundles_$environment/1-$k8s_version.yaml"
+  packages=$(yq e '.packages' "$k8s_yaml_file")
+  package_count=$(echo "$packages" | yq e 'length' -)
+
+  echo "Updating input file for kubernetes version: 1-$k8s_version"
+  # Iterate over all packages in yaml file
+  for package_index in $(seq 0 $(($package_count - 1))); do
+    package=$(echo "$packages" | yq e ".[$package_index]" -)
+    org=$(echo "$package" | yq e '.org' -)
+    projects=$(echo "$package" | yq e '.projects' -)
+    project_count=$(echo "$projects" | yq e 'length' -)
+
+    # Iterate over each project in org
+    for project_index in $(seq 0 $((project_count - 1))); do
+      project=$(echo "$projects" | yq e ".[$project_index]" -)
+
+      # Extract registry and repository from each project
+      registry=$(echo "$project" | yq e '.registry' -)
+      repository=$(echo "$project" | yq e '.repository' -)
+
+      # Get the latest version for the repository
+      latest_tag=$(get_latest_version $repository $k8s_version)
+
+      if [ "$latest_tag" == "None" ]; then
+        echo "No tags found for repository: $repository in registry $registry. Skipping..."
+        continue
+      fi
+
+      yq e -i ".packages[$package_index].projects[$project_index].versions[0].name |= \"$latest_tag\"" "$k8s_yaml_file"
+      echo "Updated $org/$repository to latest_tag $latest_tag"
+    done
+  done
+}
+
+function main() {
+  local environment=""
+  # Parse command line options
+  while getopts "e:" opt; do
+    case ${opt} in
+    e)
+      environment=$OPTARG
+      ;;
+    \?) # Invalid Option
+      usage
+      exit
+      ;;
+    esac
+  done
+
+  # Check if environment is provided and valid
+  if [[ -z "$environment" ]]; then
+    echo "Error: Environment not specified" >&2
+    usage
+  fi
+
+  if [[ "$environment" != "staging" && "$environment" != "prod" ]]; then
+    echo "Error: Invalid environment. Use 'staging' or 'prod'" >&2
+    usage
+  fi
+
+  # Check if required tools(aws,jq,yq) are installed
+  check_requirements
+
+  # Generate bundle files for Kubernetes versions 1.27 to 1.31 for given environment
+  for version in {27..31}; do
```

Comment:
We should ideally get this from a config file or from folder structure instead of hardcoded

Reply:
Updated, getting latest supported release from [here](https://raw.githubusercontent.com/aws/eks-anywhere-build-tooling/refs/heads/main/release/SUPPORTED_RELEASE_BRANCHES)